### DISCUSSION TITLE
`run_layer_method` not working with multiple args for `ui.leaflet`

### DISCUSSION THREAD

**Discussion by AIBappa** [zauberzeug/nicegui#3954] [Q&A] [answered]
Below sample code works for `bindPopup` method but not for `marker.on('click', function(e){..})` method from leaflet.js.

```py
@ui.page('/')
async def page():
    map = ui.leaflet(center=(51.5, -0.1))
    m = map.marker(latlng=map.center)
    await map.initialized()
    map.run_layer_method(m.id, 'bindPopup', 'London')
    map.run_layer_method(m.id, ':on', "click", f"function(e) {{ alert(`Marker clicked at ${e.latlng.lat}, ${e.latlng.lng}`); }}")
```

**Comment by falkoschindler** [comment, marked as answer]
Hi @AIBappa,

This line should work:
```py
map.run_layer_method(m.id, ':on', '"click"', 'function(e) { alert(`Clicked at ${e.latlng.lat}, ${e.latlng.lng}`); }')
```

- We don't need the f-string here, because the string interpolation happens in JavaScript on the client.
- Without an f-string we need to replace `{{` and `}}` with `{` and `}`.
- Because ":on" indicates dynamic arguments that will be translated from strings to JavaScript objects on the client, the name "click" needs to be a quoted string. Otherwise JavaScript will look for an object called `click`, causing an error in the developer console.

**Reply by AIBappa** [reply]
![image](https://github.com/user-attachments/assets/104570d3-777e-4dfc-8b8d-2fd390a86e89)
Yes it does indeed run. Thank you very much!
